remoulade/state/middleware.py

```diff
@@ -0,0 +1,52 @@
+import sys
+
+from ..middleware import Middleware
+from .backend import State, StatesEnum
+
+
+class MessageState(Middleware):
+
+    """Middleware use to storage and update the state
+    of the messages.
+    Parameters
+    state_ttl(int):Time(seconds) that the state will be storage
+                    in the database
+    max_size(int): Maximun size of arguments allow to storage
+                    in the database, default 2MB
+    """
+
+    def __init__(self, backend, state_ttl=3600, max_size=2e6):
+        self.backend = backend
+        self.state_ttl = state_ttl
+        self.max_size = max_size
+
+    def save(self, message, state):
+        args = message.args
+        kwargs = message.kwargs
```

Comment:
you need to check the size of both args and kwargs:
`sys.getsizeof(args) + sys.getsizeof(kwargs)` and set kwargs to `{}`.

Reply:
I modified just a little bit just in case, one of them is not too big to show it:
 if sys.getsizeof(args) then max_size then args = []
 if sys.getsizeof(kargs) then max_size then kwargs = {}